itwin-platform-access/imodels-access-backend/src/BackendIModelsAccess.ts

```diff
@@ -254,6 +254,21 @@ export class BackendIModelsAccess implements BackendHubAccess {
     return rangeTotalBytes;
   }
 
+  private async queryCurrentOrPrecedingV2Checkpoint(arg: CheckpointProps): Promise<V2CheckpointAccessProps | undefined> {
+    const getSingleChangesetParams: GetSingleChangesetParams = {
+      ...this.getIModelScopedOperationParams(arg),
+      ...PlatformToClientAdapter.toChangesetIdOrIndex(arg.changeset)
+    }
+
+    const changeset = await this._iModelsClient.changesets.getSingle(getSingleChangesetParams);
+
+    const checkpoint = await changeset.getCurrentOrPrecedingCheckpoint();
```

Comment:
If we're querying changeset index 100 and a v1 checkpoint exists at index 99(but not a v2 checkpoint) , but a v2 checkpoint exists at index 98. I'm guessing we'll get the v1 checkpoint at index 99, is that correct? Is there a way we can ask for v2 checkpoints only?

Reply:
Yes, that is correct, the API returns the latest checkpoint regardless of the checkpoint version. Does the latest iTwin.js do anything with V1 Checkpoints? Are they still used as a fallback or are they ignored completely?

There is no way to ask the iModels API for current or preceding checkpoint that is specifically a V2 Checkpoint. That logic would have to be implemented on the client and I think it could be something like this:
````
function findCurrentOrPrecedingV2Checkpoint(changesetIndex: number) {
    if (changesetIndex <= 0)
        return undefined;

    const changeset = await this._iModelsClient.changesets.getSingle(getSingleChangesetParams);
    const checkpoint = await changeset.getCurrentOrPrecedingCheckpoint();

    if (!checkpoint)
        return undefined;

    if (!!checkpoint.containerAccessInfo)
        return checkpoint.containerAccessInfo;

    const previousChangesetIndex = checkpoint.changesetIndex - 1;
    return findCurrentOrPrecedingV2Checkpoint(previousChangesetIndex);
}


```

Reply:
Thanks for the suggestion, I added that in. 

Yes latest iTwin.js will fall back to downloading a v1 checkpoint if the attempt to download a v2 checkpoint fails with an error code of IModelStatus.NotFound.